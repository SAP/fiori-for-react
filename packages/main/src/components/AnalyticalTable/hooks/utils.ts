export const makeTemplateColumns = (columns, resizedColumns) =>
  columns
    .map((column) => {
      if (resizedColumns[column.id]) {
        return `${resizedColumns[column.id]}px`;
      }
      return `minmax(${column.minWidth}px, ${column.width || '1fr'})`;
    })
    .join(' ');
