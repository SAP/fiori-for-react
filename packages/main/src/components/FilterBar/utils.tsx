import { cloneElement, ReactElement } from 'react';

export const setPropsOfChildren = (children, currentRef) => {
  return children.map((child) => {
    if (child.props.children.props[currentRef]?.children) {
      const childrenOfChild = child.props.children;
      if (
        child.props.children.props[currentRef]?.tagName === 'UI5-SELECT' ||
        child.props.children.props[currentRef]?.tagName === 'UI5-MULTI-COMBOBOX'
      ) {
        const selectedIndices = Array.from(child.props.children.props[currentRef].children)
          .map((item, index) => (item._state.selected ? index : false))
          .filter((el) => el !== false);
        const options = child.props.children.props.children.map((item, index) => {
          if (selectedIndices.includes(index)) {
            return cloneElement(item, { selected: true });
          }
          return cloneElement(item, { selected: false });
        });
        return cloneElement(child, {
          children: { ...childrenOfChild, props: { ...childrenOfChild.props, children: options } }
        });
      }
      if (
        child.props.children.props[currentRef]?.tagName === 'UI5-INPUT' ||
        child.props.children.props[currentRef]?.tagName === 'UI5-DATEPICKER'
      ) {
        return cloneElement(child, {
          children: {
            ...childrenOfChild,
            props: { ...childrenOfChild.props, value: child.props.children.props[currentRef]._state.value }
          }
        });
      }
      if (
        child.props.children.props[currentRef]?.tagName === 'UI5-SWITCH' ||
        child.props.children.props[currentRef]?.tagName === 'UI5-CHECKBOX'
      ) {
        return cloneElement(child, {
          children: {
            ...childrenOfChild,
            props: { ...childrenOfChild.props, checked: child.props.children.props[currentRef]._state.checked }
          }
        });
      }
      return child;
    }
    return child;
  });
};

export const addRef = (children, refsArray, currentRef) =>
  children.map((child) => {
    const childRef = refsArray.filter((item) => item.key === child.key);
    const childrenAttributes = child.props.children;
    const childrenProps = child.props.children.props;
    return cloneElement(child as ReactElement<any>, {
      children: { ...childrenAttributes, props: { ...childrenProps, [currentRef]: childRef[0]?.node } }
    });
  });

export const renderSearchWithValue = (searchElement, searchValue) => {
  return cloneElement(searchElement(), { value: searchValue });
};
