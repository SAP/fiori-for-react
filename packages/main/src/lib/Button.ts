import { Button, ButtonPropTypes } from '../webComponents/Button';

export { Button, ButtonPropTypes };
