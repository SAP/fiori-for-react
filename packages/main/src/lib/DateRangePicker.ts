import { DateRangePicker } from '../webComponents/DateRangePicker';

export { DateRangePicker };
