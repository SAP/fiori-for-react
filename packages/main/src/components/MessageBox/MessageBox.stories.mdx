import { Title, Subtitle, Description, Meta, Story, Canvas, ArgsTable } from '@storybook/addon-docs/blocks';
import { DocsHeader } from '@shared/stories/DocsHeader';
import { Button } from '@ui5/webcomponents-react/lib/Button';
import { MessageBox } from '@ui5/webcomponents-react/lib/MessageBox';
import { MessageBoxActions } from '@ui5/webcomponents-react/lib/MessageBoxActions';
import { MessageBoxTypes } from '@ui5/webcomponents-react/lib/MessageBoxTypes';
import { useRef, useCallback } from 'react';
import { createSelectArgTypes } from '@shared/stories/createSelectArgTypes';

<Meta
  title="Components / MessageBox"
  component={MessageBox}
  argTypes={{ ref: { type: null }, ...createSelectArgTypes({ type: MessageBoxTypes }) }}
  args={{
    open: false,
    type: MessageBoxTypes.CONFIRM,
    children: 'Press "Escape" to close the MessageBox.'
  }}
/>

<DocsHeader />

<br />

### Opening MessageBoxes

`MessageBoxes` can only be opened by attaching a `ref` to the component and then call the corresponding **`open`** method.

```JSX
const MessageBoxComponent = () => {
  const messageBoxRef = useRef();
  const onButtonClick = () => {
    messageBoxRef.current.open();
  };
  return (
    <>
      <Button onClick={onButtonClick}>Open MessageBox</Button>
      <MessageBox ref={messageBoxRef}>Some Content</MessageBox>
    </>
  );
};
```

<Canvas>
  <Story name="Default">
    {(args) => {
      const messageBoxRef = useRef();
      const onButtonClick = (e) => {
        messageBoxRef.current.open();
      };
      const onCloseMessageBox = (e) => {
        args.onClose(e);
        messageBoxRef.current.close();
      };
      return (
        <>
          <Button onClick={onButtonClick}>Open Messagebox</Button>
          <MessageBox {...args} ref={messageBoxRef} onClose={onCloseMessageBox} />
        </>
      );
    }}
  </Story>
</Canvas>

<ArgsTable story="." />

### Using MessageBoxes inside other components

`MessageBoxes` are often used within other components, when opened this could sometimes have unwanted side effects.
In this case, we recommend using [createPortal](https://reactjs.org/docs/portals.html) to mount the `MessageBox` outside of the DOM hierarchy of the parent component.

```JSX
const MessageBoxComponent = () => {
  const messageBoxRef = useRef();
  const onButtonClick = () => {
    messageBoxRef.current.open();
  };
  return (
    <>
      <Button onClick={onButtonClick}>Open MessageBox</Button>
      {createPortal(
        <MessageBox ref={messageBoxRef}>Some Content</MessageBox>,
        document.body
      )}
    </>
  );
};
```

### Closing MessageBoxes

Closing `MessageBoxes` works in the same way as opening them. So again an attached `ref` is needed on which the corresponding `close` method is called.
You can use the `onClose` prop of the `MessageBox` to close it by clicking on an action button.

```JSX
const MessageBoxComponent = () => {
  const messageBoxRef = useRef();
  const handleOpen = (e) => {
    messageBoxRef.current.open();
  };
  const handleClose = () => {
    messageBoxRef.current.close();
  };
  return (
    <>
      <Button onClick={handleOpen}>Open MessageBox</Button>
      <MessageBox ref={messageBoxRef} onClose={handleClose}>
        Some Content
      </MessageBox>
    </>
  );
};
```

<br />

# Stories

<br />

## MessageBox with custom actions

<Canvas>
  <Story name="With Custom Action">
    {(args) => {
      const messageBoxRef = useRef();
      const onButtonClick = (e) => {
        messageBoxRef.current.open();
      };
      const onCloseMessageBox = (e) => {
        messageBoxRef.current.close();
      };
      return (
        <>
          <Button onClick={onButtonClick}>Open Messagebox</Button>
          <MessageBox
            ref={messageBoxRef}
            type={args.type}
            open={args.open}
            onClose={onCloseMessageBox}
            title={args.title}
            actions={[MessageBoxActions.OK, 'A custom action', MessageBoxActions.CANCEL]}
          >
            {args.children}
          </MessageBox>
        </>
      );
    }}
  </Story>
</Canvas>
