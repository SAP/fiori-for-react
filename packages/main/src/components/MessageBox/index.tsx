import '@ui5/webcomponents-icons/dist/icons/hint';
import '@ui5/webcomponents-icons/dist/icons/message-error';
import '@ui5/webcomponents-icons/dist/icons/message-information';
import '@ui5/webcomponents-icons/dist/icons/message-success';
import '@ui5/webcomponents-icons/dist/icons/message-warning';
import '@ui5/webcomponents-icons/dist/icons/question-mark';
import { createComponentStyles } from '@ui5/webcomponents-react-base/lib/createComponentStyles';
import { useConsolidatedRef, useI18nBundle, usePassThroughHtmlProps } from '@ui5/webcomponents-react-base/lib/hooks';
import { enrichEventWithDetails } from '@ui5/webcomponents-react-base/lib/Utils';

import {
  MESSAGEBOX_ACTION_ABORT,
  MESSAGEBOX_ACTION_CANCEL,
  MESSAGEBOX_ACTION_CLOSE,
  MESSAGEBOX_ACTION_DELETE,
  MESSAGEBOX_ACTION_IGNORE,
  MESSAGEBOX_ACTION_NO,
  MESSAGEBOX_ACTION_OK,
  MESSAGEBOX_ACTION_RETRY,
  MESSAGEBOX_ACTION_YES
} from '@ui5/webcomponents-react/assets/i18n/i18n-defaults';
import { Button } from '@ui5/webcomponents-react/lib/Button';
import { ButtonDesign } from '@ui5/webcomponents-react/lib/ButtonDesign';
import { Dialog } from '@ui5/webcomponents-react/lib/Dialog';
import { Icon } from '@ui5/webcomponents-react/lib/Icon';
import { MessageBoxActions } from '@ui5/webcomponents-react/lib/MessageBoxActions';
import { MessageBoxTypes } from '@ui5/webcomponents-react/lib/MessageBoxTypes';
import { Text } from '@ui5/webcomponents-react/lib/Text';
import { Title } from '@ui5/webcomponents-react/lib/Title';
import { TitleLevel } from '@ui5/webcomponents-react/lib/TitleLevel';
import React, { FC, forwardRef, isValidElement, ReactNode, Ref, useCallback, useEffect, useMemo } from 'react';
import { CommonProps } from '../../interfaces/CommonProps';
import { Ui5DialogDomRef } from '../../interfaces/Ui5DialogDomRef';
import styles from './MessageBox.jss';

const actionTextMap = new Map();
actionTextMap.set(MessageBoxActions.ABORT, MESSAGEBOX_ACTION_ABORT);
actionTextMap.set(MessageBoxActions.CANCEL, MESSAGEBOX_ACTION_CANCEL);
actionTextMap.set(MessageBoxActions.CLOSE, MESSAGEBOX_ACTION_CLOSE);
actionTextMap.set(MessageBoxActions.DELETE, MESSAGEBOX_ACTION_DELETE);
actionTextMap.set(MessageBoxActions.IGNORE, MESSAGEBOX_ACTION_IGNORE);
actionTextMap.set(MessageBoxActions.NO, MESSAGEBOX_ACTION_NO);
actionTextMap.set(MessageBoxActions.OK, MESSAGEBOX_ACTION_OK);
actionTextMap.set(MessageBoxActions.RETRY, MESSAGEBOX_ACTION_RETRY);
actionTextMap.set(MessageBoxActions.YES, MESSAGEBOX_ACTION_YES);

export interface MessageBoxPropTypes extends CommonProps {
  open?: boolean;
  title?: string;
  children: string;
  actions?: MessageBoxActions[];
  icon?: ReactNode;
  type?: MessageBoxTypes;
  onClose: (event: CustomEvent<{ action: MessageBoxActions }>) => void;
}

const useStyles = createComponentStyles(styles, { name: 'MessageBox' });

/**
 * <code>import { MessageBox } from '@ui5/webcomponents-react/lib/MessageBox';</code>
 */
const MessageBox: FC<MessageBoxPropTypes> = forwardRef((props: MessageBoxPropTypes, ref: Ref<Ui5DialogDomRef>) => {
  const { open, type, children, className, style, tooltip, slot, title, icon, actions, onClose } = props;

  const i18nBundle = useI18nBundle('@ui5/webcomponents-react');

  const classes = useStyles();

  const iconToRender = useMemo(() => {
    if (isValidElement(icon)) return icon;
    switch (type) {
      case MessageBoxTypes.CONFIRM:
        return <Icon name="question-mark" />;
      case MessageBoxTypes.ERROR:
        return <Icon name="message-error" />;
      case MessageBoxTypes.INFORMATION:
        return <Icon name="message-information" />;
      case MessageBoxTypes.SUCCESS:
        return <Icon name="message-success" />;
      case MessageBoxTypes.WARNING:
        return <Icon name="message-warning" />;
      case MessageBoxTypes.HIGHLIGHT:
        return <Icon name="hint" />;
    }

    return null;
  }, [icon, type]);

  const titleToRender = useMemo(() => {
    if (title) {
      return title;
    }
    switch (type) {
      case MessageBoxTypes.CONFIRM:
        return 'Confirmation';
      case MessageBoxTypes.ERROR:
        return 'Error';
      case MessageBoxTypes.INFORMATION:
        return 'Information';
      case MessageBoxTypes.SUCCESS:
        return 'Success';
      case MessageBoxTypes.WARNING:
        return 'Warning';
      case MessageBoxTypes.HIGHLIGHT:
        return 'Highlight';
    }
    return null;
  }, [title, type]);

  const actionsToRender = useMemo(() => {
    if (actions && actions.length > 0) {
      return actions;
    }
    if (type === MessageBoxTypes.CONFIRM) {
      return [MessageBoxActions.OK, MessageBoxActions.CANCEL];
    }
    if (type === MessageBoxTypes.ERROR) {
      return [MessageBoxActions.CLOSE];
    }
    return [MessageBoxActions.OK];
  }, [actions, type]);

  const handleOnClose = useCallback(
    (e) => {
      const { action } = e.target.dataset;
      onClose(enrichEventWithDetails(e, { action }));
    },
    [onClose]
  );

  const dialogRef = useConsolidatedRef<Ui5DialogDomRef>(ref);

  useEffect(() => {
    if (dialogRef.current) {
      if (open) {
        dialogRef.current.open();
      } else {
        dialogRef.current.close();
      }
    }
  }, [open, dialogRef]);

  const passThroughProps = usePassThroughHtmlProps(props, ['onClose']);

  return (
    <Dialog
      slot={slot}
      ref={dialogRef}
      style={style}
      tooltip={tooltip}
      className={className}
      onAfterClose={open ? handleOnClose : null}
      header={
        <header className={classes.header} data-type={type}>
          {!!iconToRender && <div className={classes.icon}>{iconToRender}</div>}
          <Title level={TitleLevel.H5}>{titleToRender}</Title>
        </header>
      }
      footer={
        <footer className={classes.footer}>
          {actionsToRender.map((action, index) => {
            const text = i18nBundle.getText(actionTextMap.get(action));
            return (
              <Button
                style={{
                  minWidth: '4rem'
                }}
                key={action}
                design={index === 0 ? ButtonDesign.Emphasized : ButtonDesign.Transparent}
                onClick={handleOnClose}
                data-action={action}
              >
                {text}
              </Button>
            );
          })}
        </footer>
      }
      {...passThroughProps}
    >
      <Text className={classes.content}>{children}</Text>
    </Dialog>
  );
});

MessageBox.displayName = 'MessageBox';

MessageBox.defaultProps = {
  open: false,
  title: null,
  icon: null,
  type: MessageBoxTypes.CONFIRM,
  actions: []
};

export { MessageBox };
