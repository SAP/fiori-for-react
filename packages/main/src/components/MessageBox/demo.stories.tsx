import { MessageBox } from '@ui5/webcomponents-react/lib/MessageBox';
import { MessageBoxTypes } from '@ui5/webcomponents-react/lib/MessageBoxTypes';
import { MessageBoxActions } from '@ui5/webcomponents-react/lib/MessageBoxActions';
import React from 'react';
import { createSelectArgTypes } from '@shared/stories/createSelectArgTypes';

export const defaultStory = (props) => (
  <MessageBox type={props.type} open={props.open} onClose={props.onClose} title={props.title}>
    Message Box Content
  </MessageBox>
);

defaultStory.storyName = 'Default';

export const withCustomActions = (props) => (
  <MessageBox
    type={props.type}
    open={props.open}
    onClose={props.onClose}
    title={props.title}
    actions={[MessageBoxActions.OK, 'A custom action', MessageBoxActions.CANCEL]}
  >
    Message Box Content
  </MessageBox>
);

withCustomActions.args = {
  title: 'A Custom Message Box Title'
};

export default {
  title: 'Components / MessageBox',
  component: MessageBox,
  argTypes: {
    ...createSelectArgTypes({ type: MessageBoxTypes }),
    ref: {
      type: null
    }
  },
  args: {
    open: true,
    type: MessageBoxTypes.CONFIRM
  }
};
