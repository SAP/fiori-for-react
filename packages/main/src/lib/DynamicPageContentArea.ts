import { DynamicPageContentArea } from '../components/DynamicPageContentArea';

export { DynamicPageContentArea };
