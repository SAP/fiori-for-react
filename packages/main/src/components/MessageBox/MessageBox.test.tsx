import { createPassThroughPropsTest, getEventFromCallback } from '@shared/tests/utils';
import { cleanStaticAreaAfterEachTest, fireEvent, render, screen } from '@tests/index';
import { MessageBox } from '@ui5/webcomponents-react/lib/MessageBox';
import { MessageBoxActions } from '@ui5/webcomponents-react/lib/MessageBoxActions';
import { MessageBoxTypes } from '@ui5/webcomponents-react/lib/MessageBoxTypes';
import { mount } from 'enzyme';
import React from 'react';
import { spy } from 'sinon';
import { act } from 'react-dom/test-utils';

describe('MessageBox', () => {
  cleanStaticAreaAfterEachTest();

  test('Confirm - OK', () => {
    const callback = jest.fn();
    render(
      <MessageBox type={MessageBoxTypes.CONFIRM} open onClose={callback}>
        Confirm
      </MessageBox>
    );

    // expect(screen.debug()).toMatchSnapshot();

    fireEvent.click(screen.getByText('OK'));

    expect(callback.mock.calls[0][0].detail.action).toEqual(MessageBoxActions.OK);
  });

  test('Confirm - Cancel', () => {
    const callback = jest.fn();
    const { container } = render(
      <MessageBox type={MessageBoxTypes.CONFIRM} open onClose={callback}>
        Confirm
      </MessageBox>
    );
    // expect(wrapper.render()).toMatchSnapshot();

    fireEvent.click(screen.getByText('Cancel'));

    expect(callback.mock.calls[0][0].detail.action).toEqual(MessageBoxActions.CANCEL);
  });

  test('Success', () => {
    const callback = jest.fn();
    render(
      <MessageBox type={MessageBoxTypes.SUCCESS} open onClose={callback}>
        Success
      </MessageBox>
    );
    // expect(wrapper.render()).toMatchSnapshot();

    fireEvent.click(screen.getByText('OK'));
    expect(callback.mock.calls[0][0].detail.action).toEqual(MessageBoxActions.OK);
  });

  test('Warning', () => {
    const callback = jest.fn();
    render(
      <MessageBox type={MessageBoxTypes.WARNING} open onClose={callback}>
        Warning
      </MessageBox>
    );
    // expect(wrapper.render()).toMatchSnapshot();

    fireEvent.click(screen.getByText('OK'));
    expect(callback.mock.calls[0][0].detail.action).toEqual(MessageBoxActions.OK);
  });

  test('Error', () => {
    const callback = jest.fn();
    render(
      <MessageBox type={MessageBoxTypes.ERROR} open onClose={callback}>
        Error
      </MessageBox>
    );
    // expect(wrapper.render()).toMatchSnapshot();

    fireEvent.click(screen.getByText('Close'));
    expect(callback.mock.calls[0][0].detail.action).toEqual(MessageBoxActions.CLOSE);
  });

  test('Information', () => {
    const callback = jest.fn();
    render(
      <MessageBox type={MessageBoxTypes.INFORMATION} open onClose={callback}>
        Information
      </MessageBox>
    );
    // expect(wrapper.render()).toMatchSnapshot();

    fireEvent.click(screen.getByText('OK'));
    expect(callback.mock.calls[0][0].detail.action).toEqual(MessageBoxActions.OK);
  });

  test('Show', () => {
    const callback = jest.fn();
    render(
      <MessageBox open onClose={callback} title="Custom" actions={[MessageBoxActions.YES, MessageBoxActions.NO]}>
        Custom
      </MessageBox>
    );
    // expect(wrapper.render()).toMatchSnapshot();

    fireEvent.click(screen.getByText('Yes'));
    expect(callback.mock.calls[0][0].detail.action).toEqual(MessageBoxActions.YES);

    fireEvent.click(screen.getByText('No'));
    expect(callback.mock.calls).toHaveLength(2);
    expect(callback.mock.calls[1][0].detail.action).toEqual(MessageBoxActions.NO);
  });

  test('Success w/ custom title', () => {
    const callback = jest.fn();
    render(
      <MessageBox type={MessageBoxTypes.SUCCESS} open onClose={callback} title="Custom Success">
        Custom Success
      </MessageBox>
    );
    // expect(wrapper.render()).toMatchSnapshot();

    fireEvent.click(screen.getByText('OK'));
    expect(callback.mock.calls[0][0].detail.action).toEqual(MessageBoxActions.OK);
  });

  test('Not open', () => {
    const callback = jest.fn();
    const wrapper = mount(
      <MessageBox type={MessageBoxTypes.SUCCESS} open={false} onClose={callback} title="Custom Success">
        Custom Success
      </MessageBox>
    );
    expect(wrapper.render()).toMatchSnapshot();
  });

  test('No Title', () => {
    const callback = jest.fn();
    render(
      <MessageBox open onClose={callback}>
        No Title
      </MessageBox>
    );
    expect(screen.getByText('Confirmation')).toBeInTheDocument();
  });

  test('Custom Action Text', async () => {
    const onClose = jest.fn();
    render(
      <MessageBox type={MessageBoxTypes.CONFIRM} actions={[MessageBoxActions.OK, 'My Custom Action']} onClose={onClose}>
        My Message Box Content
      </MessageBox>
    );

    const textOK = screen.getByText(MessageBoxActions.OK);
    expect(textOK).toBeInTheDocument();
    let customAction = screen.getByText('My Custom Action');
    expect(customAction).toBeInTheDocument();

    fireEvent.click(textOK);
    fireEvent.click(customAction);

    expect(onClose.mock.calls[0][0].detail.action).toEqual(MessageBoxActions.OK);
    expect(onClose.mock.calls[1][0].detail.action).toEqual('My Custom Action');
  });

  createPassThroughPropsTest(MessageBox);
});
