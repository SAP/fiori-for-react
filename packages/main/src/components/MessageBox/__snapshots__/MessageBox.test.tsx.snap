// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageBox Not open 1`] = `
<ui5-dialog>
  <header
    class="MessageBox-header-0"
    data-type="Success"
    slot="header"
  >
    <ui5-icon
      name="message-success"
    />
    <ui5-title
      level="H5"
    >
      Custom Success
    </ui5-title>
  </header>
  <footer
    class="MessageBox-footer-0"
    slot="footer"
  >
    <ui5-button
      data-action="OK"
      design="Emphasized"
    >
      OK
    </ui5-button>
  </footer>
  <span
    class="Text-text-0 MessageBox-content-0"
  >
    Custom Success
  </span>
</ui5-dialog>
`;
